pallets/subtensor/src/weights.rs

```diff
@@ -1,8 +1,62 @@
 use super::*;
 use crate::math::*;
+use sp_core::H256;
+use sp_runtime::traits::{BlakeTwo256, Hash};
+use sp_runtime::SaturatedConversion;
 use sp_std::vec;
 
 impl<T: Config> Pallet<T> {
+    pub fn do_commit_weights(
+        origin: T::RuntimeOrigin,
+        netuid: u16,
+        commit_hash: H256,
+    ) -> DispatchResult {
+        let who = ensure_signed(origin)?;
+        ensure!(Self::can_commit(netuid, &who), Error::<T>::CommitNotAllowed);
+
+        WeightCommits::<T>::insert(
+            netuid,
+            &who,
+            (commit_hash, Self::get_current_block_as_u64()),
+        );
+        Ok(())
+    }
+
+    pub fn do_reveal_weights(
+        origin: T::RuntimeOrigin,
+        netuid: u16,
+        uids: Vec<u16>,
+        values: Vec<u16>,
+        version_key: u64,
+    ) -> DispatchResult {
+        let who = ensure_signed(origin.clone())?;
+        let nonce: u64 = frame_system::Pallet::<T>::account(&who)
+            .nonce
+            .saturated_into();
+
+        WeightCommits::<T>::try_mutate_exists(netuid, &who, |maybe_commit| -> DispatchResult {
+            let (commit_hash, commit_block) =
+                maybe_commit.take().ok_or(Error::<T>::NoCommitFound)?;
+
+            ensure!(
+                Self::is_reveal_block_range(commit_block),
+                Error::<T>::InvalidRevealTempo
+            );
+
+            let provided_hash: H256 = BlakeTwo256::hash_of(&(
+                who.clone(),
+                nonce,
```

Comment:
if you include the account nonce in the hash, is the caller allowed to do any transaction between the commit and the reveal ?

Reply:
We have decided to remove the nonce from the hash. Thank you for contributing.